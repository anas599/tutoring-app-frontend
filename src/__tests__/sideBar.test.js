import '../matchMedia';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import { useDispatch } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import * as reactRedux from 'react-redux';
import SideBar from '../components/side-nav/sideBar';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

describe('SideNav', () => {
  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');

  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
  });

  test('renders the Navigation menu', () => {
    useDispatch.mockReturnValue(jest.fn());

    useSelectorMock.mockReturnValue({
      // Mock the values you expect from useSelector
      user: { name: 'John Doe' },
      loggedIn: true,
    });

    render(
      <BrowserRouter>
        <SideBar />
      </BrowserRouter>,
    );

    const logoText = screen.getByText('Tutoring');
    const homeLink = screen.getByText('Home');
    const reservationsLink = screen.getByText('Reservations');
    const myReservationsLink = screen.getByText('My Reservations');
    const addClassLink = screen.getByText('Add Class');
    const deleteClasseLink = screen.getByText('Delete Class');

    expect(logoText).toBeInTheDocument();
    expect(homeLink).toBeInTheDocument();
    expect(reservationsLink).toBeInTheDocument();
    expect(myReservationsLink).toBeInTheDocument();
    expect(addClassLink).toBeInTheDocument();
    expect(deleteClasseLink).toBeInTheDocument();
  });
});
