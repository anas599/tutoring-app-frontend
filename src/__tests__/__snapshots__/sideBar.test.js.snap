// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav renders the Navigation menu 1`] = `undefined`;
